include/MeshKernel/CurvilinearGridSmoothing.hpp

```diff
@@ -26,82 +26,54 @@
 //------------------------------------------------------------------------------
 
 #pragma once
+
 #include <memory>
 
 #include <MeshKernel/CurvilinearGrid.hpp>
+#include <MeshKernel/CurvilinearGridAlgorithm.hpp>
 #include <MeshKernel/Entities.hpp>
 
 namespace meshkernel
 {
 
     /// @brief A class implementing the curvilinear grid smoothing algorithm
-    class CurvilinearGridSmoothing
+    class CurvilinearGridSmoothing : public CurvilinearGridAlgorithm
     {
     public:
         /// @brief Class constructor
         /// @param[in] grid                        The input curvilinear grid
         /// @param[in] smoothingIterations         The number of smoothing iterations to perform
-        CurvilinearGridSmoothing(std::shared_ptr<CurvilinearGrid> grid,
-                                 size_t smoothingIterations);
+        CurvilinearGridSmoothing(std::shared_ptr<CurvilinearGrid> grid, size_t smoothingIterations);
 
         /// @brief Compute curvilinear grid block smoothing (modifies the m_grid nodal values)
-        void Compute();
+        /// @return The smoothed grid
+        std::shared_ptr<CurvilinearGrid> Compute() override;
 
         /// @brief Compute curvilinear grid line smoothing. The algorithm smooths the grid along the direction specified by the line.
-        /// The line must be an m or n grid line of the curvilinear grid.
-        /// The grid is smoothed in the area specified by the line in one direction and by the two corner points on the other direction.
-        /// @param[in] firstSegmentNode The first point of the line
-        /// @param[in] secondSegmentNode The second point of the line
+        /// The line must be an m_m or m_n grid line of the curvilinear grid.
         /// @param[in] lowerLeftCornerSmoothingArea The left point of the smoothing area
         /// @param[in] upperRightCornerSmootingArea The right point of the smoothing area
-        void ComputedDirectionalSmooth(Point const& firstSegmentNode,
-                                       Point const& secondSegmentNode,
-                                       Point const& lowerLeftCornerSmoothingArea,
-                                       Point const& upperRightCornerSmootingArea);
-
-        /// @brief Sets the smoothing block
-        /// @param[in] firstCornerPoint The first point defining the smoothing area
-        /// @param[in] secondCornerPoint The second point defining the smoothing area
-        void SetBlock(Point const& firstCornerPoint, Point const& secondCornerPoint);
+        /// @return The smoothed grid
+        std::shared_ptr<CurvilinearGrid> Compute(Point const& lowerLeftCornerSmoothingArea,
```

Comment:
this class has 2 methods called "compute" but in one of them you can specify a bounding box. If I hadn't read the doxygen comment I would have assumed that the second "compute" function does the same as the first one, but only within the bounding box.  But it turns out that it not the case. Maybe you could rename the compute function with the bounding box to something else?